src/addons/transitions/ReactTransitionGroup.js

```diff
@@ -17,18 +17,28 @@ var ReactTransitionChildMapping = require('ReactTransitionChildMapping');
 var assign = require('Object.assign');
 var emptyFunction = require('emptyFunction');
 
+function _defaultRender(childrenToRender) {
```

Comment:
I was trying to avoid branching, but the `bind` definitely felt dirty to me as well, so I am glad that you found it to be disagreeable as well. The branching implementation you proposed definitely assumes less.